src/cli/self_update/windows.rs

```diff
@@ -1,11 +1,16 @@
 use std::env::{consts::EXE_SUFFIX, split_paths};
 use std::ffi::{OsStr, OsString};
 use std::fmt;
-use std::io::{self, Write};
+use std::io::Write;
 use std::os::windows::ffi::{OsStrExt, OsStringExt};
 use std::path::Path;
 use std::process::Command;
-use std::sync::{Arc, LockResult, Mutex, MutexGuard};
+use std::sync::{Arc, Mutex};
+
```

Comment:
Here and in other PRs, I would prefer that we keep imports with compilation guards in the block that they belong to (so in this case, keeping them with the `std` imports). That also has the advantage that it keeps non-conditional imports and conditional imports from the same module together.

Reply:
@djc Would you mind elaborating on this? Maybe give an example? There are many different import styles being used in the project and it's hard to say which one is canonical... Of course we can change that from now on.

Reply:
@djc You could even change the style guide in the dev guide book to solidify the new style: for the moment being it's just `std-extern-self` and nothing more.

Reply:
I just mean you're adding empty lines between imports that all come from `std`. I would make this change as follows:

```rust
use std::env::{consts::EXE_SUFFIX, split_paths};
use std::ffi::{OsStr, OsString};
use std::fmt;
#[cfg(any(test, feature = "test"))]
use std::io;
use std::io::Write;
use std::os::windows::ffi::{OsStrExt, OsStringExt};
use std::path::Path;
use std::process::Command;
use std::sync::{Arc, Mutex};
#[cfg(any(test, feature = "test"))]
use std::sync::{LockResult, MutexGuard};

use anyhow::{anyhow, Context, Result};
// etc
```

and 

```rust
use winreg::enums::{RegType, HKEY_CURRENT_USER, KEY_READ, KEY_WRITE};
#[cfg(any(test, feature = "test"))]
use winreg::types::{FromRegValue, ToRegValue};
use winreg::{RegKey, RegValue};
```

(And there's a pre-existing issue here where the winreg imports really blong in the extern block.)